### ISSUE TITLE
Pruned model (ResNet-50 and MobileNetV1) finetune

### ISSUE THREAD

**Issue by BlossomingL** [anonymous47823493/EagleEye#21]
Hi~
I'm finetuning resnet50 (Pruned  50%, generated by your scripts) according my search parameters, this is my scripts and tensorboard, training epoch is 60 at present (total 120 )but the accuracy is low (0.58, but 0.742 in your paper), is anything wrong?
```python
python3 finetune.py \
--model_name resnet50 \
--num_classes 1000 \
--checkpoint models/ckpt/imagenet_resnet50_full_model.pth \
--gpu_ids 0 \
--batch_size 128 \
--dataset_path /home/linx/dataset/ImageNet2012 \
--dataset_name imagenet \
--exp_name resnet50_50flops_3 \
--search_result search_results/pruning_strategies_resnet50.txt \
--strategy_id 49 \
--epoch 120 \
--lr 1e-2 \
--weight_decay 5e-4 \
--compress_schedule_path compress_config/res50_imagenet.yaml
```
strategy index:49, score:0.133  

![ tensorboard](https://github.com/BlossomingL/leetcode/blob/master/images/resnet50.png)

**Comment by Bowenwu1** [comment]
The eval score is OK. May you need to set `--weight_decay=1e-4`.

The scripts used for Res50 will be released soon. I am too busy to do that recently. Sorry for that.

**Comment by BlossomingL** [comment]
Thanks for your quick reply, the accuracy have reached 0.74, great!

**Comment by bezorro** [comment]
> Thanks for your quick reply, the accuracy have reached 0.74, great!

Great, how many epochs did you train for?

**Comment by BlossomingL** [comment]
120

**Comment by BlossomingL** [comment]
> > Thanks for your quick reply, the accuracy have reached 0.74, great!
> 
> Great, how many epochs did you train for?

120

**Comment by bezorro** [comment]
It's weird, what about the FLOPs reduction?
The 50% FLOPs ResNet-50 should reach the accuracy of 0.764 according to our [experiments](https://github.com/anonymous47823493/EagleEye#1-adaptive-bn-based-searching-for-pruning-strategy).

**Comment by BlossomingL** [comment]
1G?

**Comment by bezorro** [comment]
> 1G? I set the 'flops_target' to 0.5 when I run search.py. Is anything wrong?

The full ResNet-50 is 4G. So the FLOPs target would be 2G if you set the `flops_target` to 0.5, the accracy should reach 0.76+
You can try [this strategy](https://github.com/anonymous47823493/EagleEye/blob/master/search_results/best_strategy_res50_50flops.txt) to see if you can reach 0.764

**Comment by BlossomingL** [comment]
> > 1G? I set the 'flops_target' to 0.5 when I run search.py. Is anything wrong?
> 
> The full ResNet-50 is 4G. So the FLOPs target would be 2G if you set the `flops_target` to 0.5, the accracy should reach 0.76+
> You can try [this strategy](https://github.com/anonymous47823493/EagleEye/blob/master/search_results/best_strategy_res50_50flops.txt) to see if you can reach 0.764

Yes! I have reach 0.751, it may have something to do with 'weight_decay'.

**Comment by bezorro** [comment]
@BlossomingL You finetune my [strategy](https://github.com/anonymous47823493/EagleEye/blob/master/search_results/best_strategy_res50_50flops.txt) and only reached 0.751?

**Comment by BlossomingL** [comment]
It's my own strategy.
```
strategy index:49, score:0.133
strategy index:317, score:0.132
strategy index:433, score:0.126
strategy index:165, score:0.122
strategy index:94, score:0.117
```

**Comment by bezorro** [comment]
I see, so there may be something to do with `weight_decay`.

**Comment by zxDeepDiver** [comment]
Hello, this script is for single-gpu finetuning.  Does the result reported in the ECCV paper is from single-gpu training? Or, could you please give me the hyper-parameters of the multi-gpu finetuing?

> @bezorro It's my own strategy.
> 
> ```
> strategy index:49, score:0.133
> strategy index:317, score:0.132
> strategy index:433, score:0.126
> strategy index:165, score:0.122
> strategy index:94, score:0.117
> 
> python3 finetune.py \
> --model_name resnet50 \
> --num_classes 1000 \
> --checkpoint models/ckpt/imagenet_resnet50_full_model.pth \
> --gpu_ids 0 \
> --batch_size 128 \
> --dataset_path /home/linx/dataset/ImageNet2012 \
> --dataset_name imagenet \
> --exp_name resnet50_50flops_3 \
> --search_result search_results/pruning_strategies_resnet50.txt \
> --strategy_id 49 \
> --epoch 120 \
> --lr 1e-2 \
> --weight_decay 5e-4 \
> --compress_schedule_path compress_config/res50_imagenet.yaml
> ```
